src/components/ContextMenu/ContextMenu.jsx

```diff
@@ -200,3 +251,116 @@ const ContextMenu = createClass({
 });
 
 export default ContextMenu;
+
+function getFlyoutPosition({
+	direction,
+	alignment,
+	targetRect,
+	flyOutHeight,
+	flyOutWidth,
+	directonOffset,
+	getAlignmentOffset,
+	alignmentOffset = alignment === ContextMenu.CENTER
+		? getAlignmentOffset(_.includes([ContextMenu.UP, ContextMenu.DOWN], direction) ? flyOutWidth : flyOutHeight)
+		: 0,
+}) {
+
+	const {
+		CENTER,
+		DOWN,
+		END,
+		LEFT,
+		RIGHT,
+		START,
+		UP,
+	} = ContextMenu;
+
+	const {
+		bottom,
+		left,
+		right,
+		top,
+		width,
+		height,
+	} = targetRect;
+
+	const {
+		clientWidth,
+	} = document.body;
+
+	const matcher = _.matches({ direction, alignment });
+
+	if (matcher({ direction: UP, alignment: START })) {
```

Comment:
it would be even more readable if it was called `matches` or `isMatch` instead of `matcher`